internal/dependency/dependency.go

```diff
@@ -118,15 +121,36 @@ func (q *QueryOptions) Merge(o *QueryOptions) *QueryOptions {
 	return &r
 }
 
+func (q *QueryOptions) Context() context.Context {
+	if q != nil && q.ctx != nil {
+		return q.ctx
+	}
+	return context.Background()
+}
+
+func (q *QueryOptions) SetContext(ctx context.Context) QueryOptions {
+	var q2 QueryOptions
+	if q != nil {
+		q2 = *q
+	}
+	q2.ctx = ctx
+	return q2
```

Comment:
this is interesting! i'm to this pattern of returning an object for a set method. i'm curious about it vs. the `q.ctx = ctx` and no return pattern.

Reply:
Again, not being the author, I came to appreciate this as well. This is very go-esque to be functional instead of imperative -- to return a copy of the change instead of updating the underlying value. `append` is common one along with other functions in the `strings` package.

Reply:
interesting! good point about it being around for functions. for me, having it on a method might be a little confusing, since it doesn't actually set context on the receiver as the name suggests. that might be just from my experience 🤔 so happy to keep it this way since there's precedence for this pattern